raiden_contracts/utils/signature.py

```diff
@@ -8,15 +8,23 @@
 
 
 def sign(privkey: str, msg: bytes, v: int = 0) -> bytes:
-    assert isinstance(msg, bytes)
-    assert isinstance(privkey, str)
+    if not isinstance(msg, bytes):
+        raise TypeError("sign(): msg is not an instance of bytes")
+    if len(msg) != 32:
+        raise ValueError("sign(): msg has to be exactly 32 bytes")
+    if not isinstance(privkey, str):
+        raise TypeError("sign(): privkey is not an instance of str")
+    if v not in {0, 27}:
+        raise ValueError(f"sign(): got v = {v} expected 0 or 27.")
 
     pk = PrivateKey.from_hex(remove_0x_prefix(privkey))
-    assert len(msg) == 32
-
     sig: bytes = pk.sign_recoverable(msg, hasher=None)
```

Comment:
Don't want to start to bikeshed here, but should we fix ` hasher = keccak256` as in the contracts? in case coincurve switches their default for some reason and we wouldn't notice in time, nodes would produce signatures which would not be accepted onchain.

Reply:
Okay I just saw you have a test for ECVerify and coincurve compatibility, so we would at least notice.

Reply:
Third thoughts: So you do the hashing on your side, which is why `hasher=None`. If you want to keep it like this, I'd change `message` to `message_hash` in the function signature. 
Also, hashing as part of signing are strongly coupled operations, so maybe moving the hashing into the sign function makes sense? Sign would than accept size wise unrestricted messages and it would be almost impossible to break compatibility between hashing and signing.

Reply:
Let me have a look in `raiden` how they do it.  The idea currently is to share the implementation first and improve the shared implementation.

Reply:
For the moment, I just change the function signature here.

Reply:
The rest will be kept track in https://github.com/raiden-network/raiden-contracts/issues/1041